### ISSUE TITLE
Implement happy eyeballs

### ISSUE THREAD

**Issue by jornane** [Amebis/eduVPN#159]
[Happy eyeballs](https://tools.ietf.org/html/rfc8305) is an algorithm to make dual stack work better. Our eduVPN servers are dual stack, and if someone has bad working IPv6, or IPv6 at the server works bad, the discovery and retrieval of tokens will error out randomly.

Related to this, we should also check if the client handles DNS roundrobin correctly, given that we want to use this for the discovery CDN.

**Comment by rozmansi** [comment]
The Windows client is using .NET Framework 4.5 for HTTPS (`System.Net.WebRequest` class). It is like this:
![image](https://user-images.githubusercontent.com/6889694/112363901-4d239680-8cd6-11eb-97c2-cf340454cedd.png)

To implement the Happy eyeballs, we'd need something more sophisticated. Like this:
![image](https://user-images.githubusercontent.com/6889694/112364014-64fb1a80-8cd6-11eb-8f39-5d29ee2624ff.png)

So, basically, we'd need to implement the complete stack from TCP/IP up anew. Maybe there's a C library for an HTTP client we can compile with MSVC (curl, wget) and provide a managed class wrappers, so it can be used from C#?

Or, wait for Microsoft to implement this in a more recent .NET Framework. Which might take decades.

**Comment by jornane** [comment]
I found this while searching for ways to incorporate curl in a C# project

https://stackoverflow.com/questions/7929013/making-a-curl-call-in-c-sharp

I see you use `HttpWebRequest`, I use `WebClient` in another project. Your first illustration captures both of these very well. But [`HttpClient`](http://msdn.microsoft.com/en-us/library/system.net.http.httpclient%28v=vs.110%29.aspx) looks promising. I'll check it out a bit myself, and I'll report back here if it looks more like the second illustration.
